### PR TITLE
Add im.gitter.Gitter

### PR DESCRIPTION
**Pull request by ghisvail**
Another application I use frequently (alongside Zulip) for online chat.

Looking forward to your comments.

### CODE REVIEW

**Comment by ghisvail** [comment]
screenshot from 2018-04-24 09-57-14

File: im.gitter.Gitter.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      "name": "gitter",
+      "buildsystem": "simple",
+      "build-commands": [
+        "ar -x gitter_3.1.0_amd64.deb",
+        "tar -xf data.tar.gz",
+        "mkdir -p /app/bin",
+        "cp -r opt/Gitter/linux64/* /app/bin",
+        "rm -f /app/bin/*.sh",
+        "install -Dm644 /app/bin/gitter.desktop /app/share/applications/gitter.desktop",
```
[the remark this answers is not in the document]
Comment (nedrichards) [reply im.gitter.Gitter.json]: You *may* need the to keep rename desktop in order to rename the icon link inside it.

**Comment by nedrichards** [comment]
bot, build im.gitter.Gitter

**Comment by nedrichards** [comment]
The appdata needs version information and screenshots. Also, it'd be great if you could send the appdata upstream.

**Comment by nedrichards** [comment]
You should also add the 32bit version, you can use only-arches as shown here: https://github.com/flathub/com.visualstudio.code/blob/master/com.visualstudio.code.json#L143 to make sure a file is only called on x86-64 or i386.

Then you need a flathub.json file that skips the other architectures, arm and aarch64. Here'a an example https://github.com/flathub/com.visualstudio.code/blob/master/flathub.json

Although, this is an open source app right? You should really be building it from source. There are helper scripts available for JS apps: https://github.com/flatpak/flatpak-builder-tools/

**Comment by ghisvail** [comment]
> Although, this is an open source app right? You should really be building it from source.

I used the same approach as with `io.atom.Atom`, is it wrong?

**Comment by nedrichards** [comment]
When atom was first added, we didn't have a way of building apps that used npm or yarn to manage their dependencies within the network free sandbox, so we had to import binaries. Now, using the tools I linked above we have a way of doing that. Maintainers should ideally move their apps to using that. Thanks for noting, I've opened a ticket there: https://github.com/flathub/io.atom.Atom/issues/9

An example of a complex tool using this tech that uses this technique would be the OSS build of VS Code: https://github.com/flathub/com.visualstudio.code.oss

**Comment by ghisvail** [comment]
I tried building `org.zulip.Zulip` from source and got the following error:
```
npm info lifecycle electron-chromedriver@1.8.0~install: electron-chromedriver@1.8.0

> electron-chromedriver@1.8.0 install /run/build/zulip/main/node_modules/electron-chromedriver
> node ./download-chromedriver.js

Downloading tmp-13-1-chromedriver-v1.8.0-linux-x64.zip
Error: getaddrinfo EAI_AGAIN github.com:443
/run/build/zulip/main/node_modules/electron-chromedriver/download-chromedriver.js:30
  if (err != null) throw err
                   ^

Error: getaddrinfo EAI_AGAIN github.com:443
    at Object._errnoException (util.js:1022:11)
    at errnoException (dns.js:55:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:92:26)
```
I suspect something similar might happen to `im.gitter.Gitter` too then?

**Comment by ghisvail** [comment]
> add a source tarball

Are there any specific instructions for that, beyond adding another entry to `sources`? Should the content of the tarball be unpacked somewhere in particular?

Glad to hear `im.gitter.Gitter` and `org.zulip.Zulip` are close to being accepted. It's been a very pleasant review experience so far.

**Comment by ghisvail** [comment]
> add a source tarball

Are there any specific instructions for that, beyond adding another entry to `sources`? Should the content of the tarball be unpacked somewhere in particular?

Glad to hear `im.gitter.Gitter` and `org.zulip.Zulip` are close to being accepted. It's been a very pleasant review experience so far.

**Comment by barthalion** [comment]
[Like this](https://github.com/flathub/im.riot.Riot/blob/master/im.riot.Riot.json#L61), without any additional steps will be enough.

**Comment by nedrichards** [comment]
bot, build im.gitter.Gitter

**Comment by ghisvail** [comment]
Is anything still actionable on my end?

**Comment by barthalion** [comment]
bot, build im.gitter.Gitter

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/im.gitter.Gitter

**Pull request closed without merging**